library/src/main/java/com/bumptech/glide/load/engine/exceptions/ErrorWrappingGlideException.java

```diff
@@ -0,0 +1,11 @@
+package com.bumptech.glide.load.engine.exceptions;
+
+/**
+ * An exception class used for wrapping and distinguishing errors such as
+ * {@link OutOfMemoryError}.
+ */
+public class ErrorWrappingGlideException extends Exception {
+    public ErrorWrappingGlideException(Throwable throwable) {
```

Comment:
I think if it's error wrapping it should expect an `Error`.

Also along the same lines maybe override getCause() with a covariant return type:
```java
public Error getCause () ...
```